main.c

```diff
@@ -168,44 +199,63 @@ int main(int argc, char **argv)
 	if (err)
 		return err;
 
-	/* Set up libbpf errors and debug info callback */
-	if (env.verbose) {
+	if (getuid() != 0) {
+		printf("ERR : you need root privileges to run this tool\n");
+		return 1;
+	}
```

Comment:
As discussed offline. This should not be needed as we're not performing any privileged operation in bpf_object__prepare().